Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/DirectoryInitializationService.java

```diff
@@ -123,16 +157,22 @@ private void copyAsset(String asset, String output, Boolean overwrite)
         }
     }
 
-    private void copyAssetFolder(String assetFolder, String outputFolder, Boolean overwrite)
+    private void copyAssetFolder(String assetFolder, File outputFolder, Boolean overwrite)
     {
         Log.verbose("[DirectoryInitializationService] Copying Folder " + assetFolder + " to " + outputFolder);
 
         try
         {
+            boolean createdFolder = false;
             for (String file : getAssets().list(assetFolder))
             {
-                copyAssetFolder(assetFolder + File.separator + file, outputFolder + File.separator + file, overwrite);
-                copyAsset(assetFolder + File.separator + file, outputFolder + File.separator + file, overwrite);
+                if (!createdFolder)
+                {
+                    outputFolder.mkdir();
+                    createdFolder = true;
+                }
+                copyAssetFolder(assetFolder + File.separator + file, new File(outputFolder, file), overwrite);
+                copyAsset(assetFolder + File.separator + file, new File(outputFolder, file), overwrite);
```

Comment:
Any thoughts on using the Java equivalent of C#'s `Path.Combine` for joining path segments? I always twitch at string concatenations for that, especially when it isn't clear whether it may (or may not) yield duplicated path separators...

Reply:
It can be done using the `File` or `Path` class (like I'm doing on the second half of these lines), but asset paths have to be strings, and I think having to convert to one of those two classes and then back to string would be a bit overkill. Or is there a way to do it in Java that only uses strings?

Reply:
Quick google tells me [`Paths.get`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Paths.html#get-java.lang.String-java.lang.String...-) might do that.

Reply:
It converts to `Path`.

Reply:
Ah, duh. Nevermind then.